File: appmy/src/main/java/com/leilei/guoshujinfu/mylearning/util/BasePresenter.java
package com.leilei.guoshujinfu.mylearning.util;

import java.util.ArrayList;
import java.util.List;

import rx.Subscription;

/**
 * @Author: AqrLei
 * @Name MyLearning
 * @Description:
 * @Date: 2017/8/7
 */

public class BasePresenter {
    protected List<Subscription> subscriptions;
    public BasePresenter() {
        subscriptions = new ArrayList<>();
    }
    public void cancle() {
        for(Subscription subscription: subscriptions) {
            if(!subscription.isUnsubscribed()) {
                subscription.unsubscribe();
            }
        }
        subscriptions.clear();

    }
    public void addSubscription(Subscription subscription) {
        subscriptions.add(subscription);
    }
}

File: yueting/src/main/kotlin/com/aqrlei/graduation/truckrental/baselib/base/BasePresenter.kt
package com.aqrlei.graduation.truckrental.baselib.base

import rx.Subscription

/**
 * @Author: AqrLei
 *@Name MyLearning
 *@Description:
 *@Date: 2017/8/22
 */
abstract class BasePresenter {
    protected var subscriptions: MutableList<Subscription>

    init {
        subscriptions = ArrayList()
    }

    fun cancle() {
        for (subscription in subscriptions) {
            if (!subscription.isUnsubscribed)
                subscription.unsubscribe()
        }
        subscriptions.clear()
    }

    fun addSubscription(subscription: Subscription) {
        subscriptions.add(subscription)
    }
}